### DISCUSSION TITLE
Error Loading App: Buddibase Client Lib

### DISCUSSION THREAD

**Discussion by miches87** [Budibase/budibase#11391] [Help] [answered]
Hello Budibase Community/Team,

it guess the problem is already know (see #3570 and #5994) but I can't find a sufficant solution.
To summarize the issue:
- Budibase App been deployed without problems
- works on most devices
- some users report a problem where the sceen shows attached error message
- it was suggested to re-deploy the app or re-build the whole app (both was done by me without solving the problem)
![IMG-20230727-WA0000](https://github.com/Budibase/budibase/assets/106280722/071315e8-8c19-4551-8e04-714df2faa590)

In #5994 was reported that there might be a problem is iOS. My users who reported the problem uses Windows.

Some information on my application:
- Form application
- running on Budibase Cloud
- latest version 2.8.27
- Google Sheet is used as connected database

Any idea how to solve it?
Best regards
Michael

**Comment by miches87** [comment]
Any idea how to solve the issue? Maybe it's a big of the platform since some other users also reported this problem in the past

**Comment by melohagan** [comment]
Hey @miches87 

I think I'll need a bit more info to have any hope of helping out! 

> My users who reported the problem uses Windows.
> works on most devices

Is it possible you can narrow this down further, can you get the specific specs of the devices that are not working? What percentage of your users are seeing this error?

**Comment by miches87** [comment]
@melohagan, sry for the late reply. I had to do some more analysis.
The problem lies in the reverse proxy configuration of the client. The proxies block the loading of the "budibase-client.js" package and another package. As a result, the browser says that the app is loading with errors.
I use Google Sheets as a database for my app. Does the browser see this connection and block the package because of this extra connection?

Since the problem occurs with different customers in different industries, I see a general problem with these packages that BudiBase uses. Any idea how I can solve the problem?
![2023-08-15 BudiBase Error message](https://github.com/Budibase/budibase/assets/106280722/76a2bd00-1263-4945-a923-551a53473c7e)

**Comment by miches87** [comment]
Hello,
any idea how to fix this issue?

**Reply by ConorWebb96** [reply, marked as answer]
Hey @miches87,

This thread will help; https://github.com/Budibase/budibase/discussions/10823#discussioncomment-6120717

**Reply by miches87** [reply]
Thanks @ConorWebb96. Sounds like a reasonable explaination and also my guess that there is a firewall issue.
Nevertheless defining firewall whitelisting with cooperate IT departments are not an easy task.
Do you have any idea why FWs have a problem with the IPs?

**Reply by ConorWebb96** [reply]
Hey @miches87,

Basically its due to not wanting any IP addresses being able to access your Budibase installation, it should only be trusted IPs for security reasons. Simply put It's just good security practices.
